tmt/steps/__init__.py

```diff
@@ -392,12 +404,145 @@ def go(self):
         self.verbose('order', self.order, 'magenta', level=3)
 
 
-class Login(tmt.utils.Common):
-    """ Log into the guest """
-
+class ActionStep(tmt.utils.Common):
     # Dictionary containing list of requested phases for each enabled step
     _phases = None
 
+    @classmethod
+    def phases(cls, step):
+        """ Return list of phases enabled for given step """
+        # Build the phase list unless done before
+        if cls._phases is None:
+            cls._phases = cls._parse_phases(step)
+        # Return enabled phases, empty list if step not found
+        try:
+            return cls._phases[step.name]
+        except KeyError:
+            return []
+
+    @classmethod
+    def phases(cls, step):
+        """ Return list of phases enabled for given step """
+        # Build the phase list unless done before
+        if cls._phases is None:
+            cls._phases = cls._parse_phases(step)
+        # Return enabled phases, empty list if step not found
+        try:
+            return cls._phases[step.name]
+        except KeyError:
+            return []
+
+    @classmethod
+    def _parse_phases(cls, step):
+        """ Parse options and store phase order """
+        phases = dict()
+        options = cls._opt('step')
+
+        # Use the end of the last enabled step if no --step given
+        if not options:
+            last_enabled_step = list(step.plan.steps())[-1]
+            phases[last_enabled_step.name] = [PHASE_END]
+            return phases
+
+        # Process provided options
+        for option in options:
+            # Parse the step:phase format
+            matched = re.match(r'(\w+)(:(\w+))?', option)
+            if matched:
+                step_name, _, phase = matched.groups()
+            if not matched or step_name not in STEPS:
+                raise tmt.utils.GeneralError(f"Invalid step '{option}'.")
+            # Check phase format, convert into int, use end by default
+            try:
+                phase = int(phase)
+            except TypeError:
+                phase = PHASE_END
+            except ValueError:
+                # Convert 'start' and 'end' aliases
+                try:
+                    phase = dict(start=PHASE_START, end=PHASE_END)[phase]
+                except KeyError:
+                    raise tmt.utils.GeneralError(f"Invalid phase '{phase}'.")
+            # Store the phase for given step
+            try:
+                phases[step_name].append(phase)
+            except KeyError:
+                phases[step_name] = [phase]
+        return phases
+
+
+class Reboot(ActionStep):
+    """ Reboot guest """
+
+    # Which kind of reboot
+    _hard = False
+
+    # True if interactive login enabled
+    _enabled = False
+
+    # guest name provided via CLI
```

Comment:
```suggestion
    # Guest name provided via CLI
```
Nitpick for consistency reasons

Reply:
Thanks. Solved.